cmd/devicesimulator/main.go

```diff
@@ -64,37 +80,59 @@ func main() {
 	log.Infoln("setting up metrics endpoint")
 	setupMetricsEndpoint(*metricsAddr)
 
+	serviceClient, err := client.NewFromConfigFile(client.DefaultFlightctlClientConfigPath())
+	if err != nil {
+		log.Fatalf("Error creating service client: %v", err)
+	}
+
 	log.Infoln("creating agents")
 	agents := make([]*agent.Agent, *numDevices)
+	agentsFolders := make([]string, *numDevices)
 	for i := 0; i < *numDevices; i++ {
 		agentName := fmt.Sprintf("device-%04d", i)
 		certDir := filepath.Join(agentConfigTemplate.ConfigDir, "certs")
 		agentDir := filepath.Join(*dataDir, agentName)
+		err := os.Setenv(client.TestRootDirEnvKey, agentDir)
```

Comment:
Added it from test_harness but i believe this os env should be deleted and paths should be provided from outside to each object and not magically set by some env in different places. 
I believe it is worth ticket in backlog for cleanup